### ISSUE TITLE
unreproducible integration test failures

### ISSUE THREAD

**Issue by btbonval** [medic/medic-analytics#4]
I've gotten a few integration test failures since the code first worked successfully. It seems to be a different failure each time a failure occurs, and usually the next run works out fine. In other words, the failures are transient.

This time the error was:
```
  1) Integration moves all documents to postgres:
     TypeError: Cannot read property 'data' of undefined
      at test/integration/index.js:62:34
```
I added code to catch this error and log its query in the future. However, after adding that code, the integration test succeeded.

**Comment by btbonval** [comment]
Keeping postgres and couchdb settings the same, I changed  `POSTGRESQL_TABLE` and `POSTGRESQL_COLUMN` to match `.travis.yml`. I get an integration test failure:
```
  1) Integration moves all documents to postgres:
     AssertionError: expected undefined to deeply equal { Object (_id, _rev, ...) }
```

The `couchdata` table is created and contains 3653 rows, the same as `nosql` (from the previous local integration test run). Yet the test fails.

I ran it twice, both failed. This issue seemed to see the first test fail but subsequent tests succeed, but it might still be related somehow.

**Comment by btbonval** [comment]
I just noticed the column under test in `.travis.yml` is called `json` which is also a datatype in PostgreSQL. I was not having problems in `psql` with that. In fact the table has the column correctly named. However, it might be causing issues.

Using table `couchdata` and column `json` does cause the integration test to fail locally, but with a different error than Travis.

I renamed that to `jsondata`, but the test still fails. The data in `couchdata` has the correct number of rows and the column name doesn't seem to matter.

If I use the table `nosql` and column `data`, the integration test passes. If I use the table `couchdata` and column `json` or `jsondata`, the integration test fails. Let's try mixing them up: `nosql` and `jsondata`, then `couchdata` and `data`.
